climada_petals/entity/exposures/osm_dataloader.py

```diff
@@ -0,0 +1,346 @@
+"""
+This file is part of CLIMADA.
+
+Copyright (C) 2017 ETH Zurich, CLIMADA contributors listed in AUTHORS.
+
+CLIMADA is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free
+Software Foundation, version 3.
+
+CLIMADA is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along
+with CLIMADA. If not, see <https://www.gnu.org/licenses/>.
+
+---
+
+Download openstreetmap data from the Overpass API
+"""
+
+import itertools
+import logging
+import time
+import geopandas as gpd
+import overpy
+import shapely
+
+LOGGER = logging.getLogger(__name__)
+
+class OSMApiQuery:
+    """
+    Queries features directly via the overpass turbo API.
+
+    area: tuple (xmin, ymin, xmax, ymax), list [xmin, ymin, xmax, ymax]
+        or shapely.geometry.Polygon
+    query: str
+        must be of format '["key"]' or '["key"="value"]', etc.
```

Comment:
This information is outdated. There is no `query` anymore, but `condition`. Notice also that the *attribute* `area` is either a tuple or a string, whereas the *parameter* `area` to the `__init__` can be a sequence or a polygon. Please make this distinction clear.

Reply:
updated it accordingly.